v3/src/models/shared/shared-case-metadata.ts

```diff
@@ -194,6 +194,31 @@ export const SharedCaseMetadata = SharedModel
       return categorySet
     }
   }))
+  .actions(self => ({
+    afterCreate() {
+      // respond to change of collections by showing hidden attributes if all attributes are hidden
+      addDisposer(self, reaction(
+        () => {
+          return {
+            hiddenAttrs: Array.from(self.hidden.keys()),
+            collections: self.data?.collections.map(collection => collection.attributes.map(attr => attr?.id)) || []
+          }
+        },
+        ({hiddenAttrs}) => {
+          self.data?.collections.forEach(collection => {
+            const attrs = collection.attributes
+            if (attrs && attrs.every(attr => attr && hiddenAttrs.includes(attr.id))) {
```

Comment:
```suggestion
            if (attrs && attrs.every(attr => attr && !!self.hidden.get(attr.id))) {
```
We don't expect the list of hidden attributes to get particularly long, but as a general rule it's good to avoid linear search through an array and the corresponding O(n^2) behavior where possible, and in this case we've got ready access to a map of hidden attributes for efficient lookup. Even though we will no longer be using the `hiddenAttrs` value returned by the accessor function, it's still worth returning it so that changes will trigger the reaction.